packages/core/src/fhir-deduplication/deduplicate-fhir.ts

```diff
@@ -339,6 +357,30 @@ function removeDanglingReferences<T extends Resource>(entry: T, link: string): T
       return section;
     });
   }
+  if ("location" in entry) {
+    if (entry.resourceType === "Encounter") {
+      entry.location = entry.location.filter(location => location.location?.reference !== link);
+      if (!entry.location.length) delete entry.location;
+    }
+  }
+  if ("participant" in entry) {
+    if (entry.resourceType === "Encounter") {
+      entry.participant = entry.participant?.filter(part => part.individual?.reference !== link);
+      if (!entry.participant.length) delete entry.participant;
+    }
+  }
```

Comment:
I think we are missing serviceProvider as a field in entry. 
See Encounters.hbs and serviceProvider.hbs